### PR TITLE
Create Go - Concepts - Strings - Terms - HasSuffix

### PR DESCRIPTION
**Pull request by Jamesllllllllll**
### Description

Adding `HasSuffix()` term for Go.
Closes [Issue #3066](https://github.com/Codecademy/docs/issues/3066)

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/go/concepts/strings/terms/hassuffix/hassuffix.md
Comment (SSwiniarski) [inline content/go/concepts/strings/terms/hassuffix/hassuffix.md:74]:
```suggestion
```

This duplicates the prior `for` loop, and the parser complains about not using `index` in it

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@Jamesllllllllll I did a first review and had one comment.

File: content/go/concepts/strings/terms/hassuffix/hassuffix.md
Comment (Jamesllllllllll) [reply content/go/concepts/strings/terms/hassuffix/hassuffix.md]: Whoops! That was a second version of the loop I was testing. I removed it and pushed my changes.

Comment (SSwiniarski) [reply content/go/concepts/strings/terms/hassuffix/hassuffix.md]: @Jamesllllllllll You can mark them resolved. Or you can click the commit suggestion button and it will commit the change and resolve it for you.

**Review by SSwiniarski — APPROVED** [review]
@Jamesllllllllll Looks good for a second review.

File: content/go/concepts/strings/terms/hassuffix/hassuffix.md
Comment (yangc95) [inline content/go/concepts/strings/terms/hassuffix/hassuffix.md:14]:
```suggestion
The **`HasSuffix()`** function in Go is used to check if a specified suffix exists on a given string. The function returns `true` if the given string has the specified suffix, otherwise it returns `false`.
```

File: content/go/concepts/strings/terms/hassuffix/hassuffix.md
Comment (yangc95) [inline content/go/concepts/strings/terms/hassuffix/hassuffix.md:22]:
```suggestion
Where `str` is the given string and `suffix` is the ending substring to check for.
```

Comment (yangc95) [reply content/go/concepts/strings/terms/hassuffix/hassuffix.md:22]: @Jamesllllllllll one last comment, LGTM after!

Comment (Jamesllllllllll) [reply content/go/concepts/strings/terms/hassuffix/hassuffix.md]: @yangc95 thank you!!

**Review by yangc95 — APPROVED** [review]

**Pull request merged**
